packages/@aws-cdk/core/lib/cfn-resource.ts

```diff
@@ -289,6 +293,45 @@ export class CfnResource extends CfnRefElement {
     addDependency(this, target, `"${Node.of(this).path}" depends on "${Node.of(target).path}"`);
   }
 
+  /**
+   * Indicates that this resource no longer depends on another resource.
+   *
+   * This can be used for resources across stacks (including nested stacks)
+   * and the dependency will automatically be removed from the relevant scope.
+   */
+  public removeDependsOn(target: CfnResource) : void {
```

Comment:
I wrote these comments out of order but kind of the same feedback at below. I'm not sure I like the <action>DependsOn phrasing of each of these.